lib/system.nim

```diff
@@ -1871,6 +1871,21 @@ proc `$` *[Enum: enum](x: Enum): string {.magic: "EnumToStr", noSideEffect.}
   ## a ``$`` operator for a concrete enumeration is provided, this is
   ## used instead. (In other words: *Overwriting* is possible.)
 
+proc newString*[N](data: array[N, char]): string {.noSideEffect.} =
```

Comment:
This procedure is for all cases where an array char is converted into a string. So it does what ``$`` on ``array[..., char]`` used to do. I personally don't like to introduce new names for methods when it's not necessary. But I can easily make this procedure work for both, null terminated and non null terminated arrays. What do you think about that?